base/docs/Docs.jl

```diff
@@ -675,4 +675,20 @@ function hasdoc(binding::Docs.Binding, sig::Type = Union{})
 end
 
 
+"""
+    undocumented_names(mod::Module; all=false)
+
+Return an array of undocumented symbols in `module` (that is, lacking docstrings).
+`all=false` returns only exported symbols; whereas `all=true` also includes
+non-exported symbols, following the behavior of [`names`](@ref). Only valid identifiers
+are included. Names are returned in sorted order.
+
+See also: [`names`](@ref), [`Docs.hasdoc`](@ref), [`Base.isidentifier`](@ref).
+"""
+function undocumented_names(mod::Module; all::Bool=false)
+    filter!(names(mod; all)) do sym
+        !hasdoc(mod, sym) && Base.isidentifier(sym)
```

Comment:
I just realized that this filters out macros `@foo`, which we don't want.  Probably should be
```jl
str = string(sym)
!hasdoc(mod, sym) && (Base.isidentifier(sym) || (startswith(str, "@") && Base.isidentifier(str[2:end])))
```